File: book/systemd/clang_systemd/helloworld.c
#include <stdio.h>
#include <time.h>
#include <unistd.h>
#include <stdlib.h>
#include <systemd/sd-daemon.h>

int main(int argc, char *argv[]) {
    printf("Starting up ...");
    sleep(10);
    printf("Startup complete before notify");

    // Send a READY message to systemd
    sd_notify(0, "READY=1");

    // Send a STATUS message to systemd
    sd_notify(0, "STATUS=status message");

    // Send a WATCHDOG message to systemd
    //sd_notify(0, "WATCHDOG=1");

    for(;;) 
	{
        printf("Hello from the Python Demo Service");
        sd_notify(0, "STATUS=Processing  data");
        sleep(5);
        sd_notify(0, "STATUS=Waiting for data");
        sleep(5);
	}
    return 0;
}


// sudo yum install systemd-devel
// gcc -o my-program helloworld.c -lsystemd
//

File: book/systemd/python_systemd/helloworld.py

if __name__ == '__main__':
    import time
    import systemd.daemon

    print('Starting up ...')
    time.sleep(5)
    print('Startup complete before notify')
    systemd.daemon.notify('READY=1')
    print('Startup complete after notify')

    while True:
        print('Hello from the Python Demo Service')
        systemd.daemon.notify("STATUS=Processing  data")
        time.sleep(5)
        systemd.daemon.notify("STATUS=Waiting for data")
        time.sleep(5)

